File: Assets/Game/Scripts/Moduls/PC_TwoFlames.lua
---@class PC_TwoFlames : ModulBase 
PC_TwoFlames = ModulBase:extend()

function PC_TwoFlames:new()
  PC_TwoFlames.super.new(self)
  self.FlameSmallLeft = nil ---@type GameObject
  self.FlameSmallRight = nil ---@type GameObject
  self.Flame = nil ---@type GameObject
  self.CheckPointTigger = nil ---@type TiggerTester
  self.CheckPointActived = false
end
function PC_TwoFlames:Start()
  self.FlameSmallLeft:SetActive(false)
  self.FlameSmallRight:SetActive(false)
  self.Flame:SetActive(false)
  
  if not self.IsPreviewMode then
    --Tigger 进入事件
    ---@param otherBody GameObject
    self.CheckPointTigger.onTriggerEnter = function (s, otherBody)
      if not self.CheckPointActived and otherBody.tag == 'Ball' then
        --触发下一关
        self.CheckPointActived = true
        self.Flame:SetActive(false)
        GamePlay.SectorManager:NextSector()
      end
    end
  end
end

function PC_TwoFlames:Active()
  self.CheckPointActived = true
  self.FlameSmallLeft:SetActive(true)
  self.FlameSmallRight:SetActive(true)
end
function PC_TwoFlames:Deactive()
  self.FlameSmallLeft:SetActive(false)
  self.FlameSmallRight:SetActive(false)
  self.Flame:SetActive(false)
end

function PC_TwoFlames:ActiveForPreview()
  self.FlameSmallLeft:SetActive(true)
  self.FlameSmallRight:SetActive(true)
  self.gameObject:SetActive(true)
end
function PC_TwoFlames:DeactiveForPreview()
  self.FlameSmallLeft:SetActive(false)
  self.FlameSmallRight:SetActive(false)
  self.gameObject:SetActive(false)
end

function PC_TwoFlames:Reset()
end
function PC_TwoFlames:Backup()
end
---设置火焰激活状态
function PC_TwoFlames:InternalActive()
  self.FlameSmallLeft:SetActive(false)
  self.FlameSmallRight:SetActive(false)
  self.Flame:SetActive(true)
  self.CheckPointActived = false
end

function CreateClass:PC_TwoFlames()
  return PC_TwoFlames()
end

File: Assets/Game/GamePlay/Scripts/Moduls/PC_TwoFlames.cs
using UnityEngine;

namespace Ballance2.Game.GamePlay.Moduls
{
  /// <summary>
  /// 检查点火焰组件
  /// </summary>
  public class PC_TwoFlames : ModulBase 
  {
    public GameObject FlameSmallLeft;
    public GameObject FlameSmallRight;
    public GameObject Flame;
    public TiggerTester CheckPointTigger;
    public bool CheckPointActived = false;

    protected override void Start()
    {
      base.Start();
      this.FlameSmallLeft.SetActive(false);
      this.FlameSmallRight.SetActive(false);
      this.Flame.SetActive(false);

      if (!IsPreviewMode) 
      {
        //Tigger 进入事件
        CheckPointTigger.onTriggerEnter = (s, otherBody) => {
          if (!CheckPointActived && otherBody.tag == "Ball")
          {
            //触发下一关
            this.CheckPointActived = true;
            this.Flame.SetActive(false);
            GamePlayManager.Instance.SectorManager.NextSector();
          }
        };
      }
    }
    public override void Active()
    {
      base.Active();
      this.CheckPointActived = true;
      this.FlameSmallLeft.SetActive(true);
      this.FlameSmallRight.SetActive(true);
    }
    public override void Deactive()
    {
      base.Deactive();
      this.FlameSmallLeft.SetActive(false);
      this.FlameSmallRight.SetActive(false);
      this.Flame.SetActive(false);
    }

    public override void ActiveForPreview()
    {
      this.FlameSmallLeft.SetActive(true);
      this.FlameSmallRight.SetActive(true);
      this.gameObject.SetActive(true);
    }
    public override void DeactiveForPreview()
    {
      this.FlameSmallLeft.SetActive(false);
      this.FlameSmallRight.SetActive(false);
      this.gameObject.SetActive(false);
    }

    //设置火焰激活状态
    public void InternalActive()
    {
      this.FlameSmallLeft.SetActive(false);
      this.FlameSmallRight.SetActive(false);
      this.Flame.SetActive(true);
      this.CheckPointActived = false;
    }
  }
}